File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // Count the frequency O(n)
        unordered_map<int,int> frequency; //key = element, value = frequency
        for (int num : nums) {
            frequency[num]++;
        }
        
        vector<pair<int,int>> pairs;
        for (const auto& p : frequency) {
            pairs.push_back({p.second, p.first});
        }

        sort(pairs.rbegin(), pairs.rend());

        vector<int> result;
        for (int i = 0; i < k; i++) {
            result.push_back(pairs[i].second);
        }
        return result;
    }
};

File: Data Structures & Algorithms/plus-one/submission-0.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        nums = digits[::-1]
        r = 1
        for i in range(len(nums)):
            if r == 1:
                nums[i] += 1
                r = 0
            if nums[i] == 10:
                nums[i] = 0
                r = 1
        if r == 1:
            nums.append(1)
        return(nums[::-1])
        

            
        